cloudbuild/nightly/scripts/python-scripts/create_cluster.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2023 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#            http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+import argparse
+
+from absl import logging
+from absl import app
+from collections.abc import Sequence
+from google.cloud import dataproc_v1 as dataproc
+
+
+def create_cluster(project_id, region, cluster_name, num_workers, dataproc_image_version,
+                   initialisation_action_script_uri):
+    """This sample walks a user through creating a Cloud Dataproc cluster
+    using the Python client library.
+
+    Args:
+        project_id (string): Project to use for creating the cluster.
+        region (string): Region where the cluster should live.
+        cluster_name (string): Name to use for creating a cluster.
+        num_workers (int): Number of workers in the cluster.
+        dataproc_image_version (string): The Dataproc Image version used to create the cluster.
+        initialisation_action_script_uri (string): Link to the initialisation script for
+         the cluster.
+    """
+    logging.info(f'Creating cluster {cluster_name} in region {region}')
+    # Create a client with the endpoint set to the desired cluster region.
+    cluster_client = dataproc.ClusterControllerClient(
+        client_options={'api_endpoint': f'{region}-dataproc.googleapis.com:443'}
+    )
+    # Create the cluster config.
+    cluster = {
+        'project_id': project_id,
+        'cluster_name': cluster_name,
+        'config': {
+            'master_config': {'num_instances': 1, 'machine_type_uri': 'n1-standard-2'},
+            'worker_config': {'num_instances': num_workers, 'machine_type_uri': 'n2-standard-4'},
+            'software_config': {
+                'image_version': dataproc_image_version,
+                'optional_components': ['FLINK']},
+            'initialization_actions': [{'executable_file': initialisation_action_script_uri}],
+            'lifecycle_config': {'auto_delete_ttl': '3600s'},
```

Comment:
Is this enough? Whats the total duration of our nightly job?

Reply:
The nightly job (including cluster creation) runs for about 31mins.